ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -230,16 +230,23 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
     final Account sender = updater.get(senderAddress);
     final long nonce = sender != null ? sender.getNonce() : 0L;
 
-    long gasLimit =
+    final long txGasLimit =
         callParams.getGasLimit() >= 0
             ? callParams.getGasLimit()
             : blockHeaderToProcess.getGasLimit();
+
+    final long simulationGasLimit;
     if (rpcGasCap > 0) {
-      gasLimit = rpcGasCap;
+      simulationGasLimit = Math.min(txGasLimit, rpcGasCap);
```

Comment:
There is a real need to make RPC gas limit above block size limit, especially on eth_call. I wonder if we cannot make eth_estimateGas using a different variante of the transaction simulator

Reply:
using different transaction simulators is doable, let me try to implement that way

Reply:
@ahamlat I have implemented your suggestion and now `eth_call` behaves differently of other call in regards of how `rpc-gas-cap` is applied